src/common/token-utils.ts

```diff
@@ -0,0 +1,62 @@
+import { RPCClient } from '@stacks/rpc-client';
+
+function handleErrorMessage(message = 'Error') {
+  return {
+    error: `This token does not conform to the fungible token trait (SIP 10)${
+      message ? `: ${message}` : '.'
+    }`,
+  };
+}
+
+export async function isSip10Transfer({
+  contractAddress,
+  contractName,
+  networkUrl,
+}: {
+  contractAddress: string;
+  contractName: string;
+  networkUrl: string;
+}): Promise<{ okay: true } | { error: string }> {
+  try {
+    const rpcClient = new RPCClient(networkUrl);
+    const contractInterface = await rpcClient.fetchContractInterface({
+      contractAddress,
+      contractName,
+    });
+    let hasCorrectName = false;
+    let hasCorrectSender = false;
+    let hasCorrectRecipient = false;
+    let hasCorrectAmount = false;
+    let hasCorrectNumberOfArgs = false;
+
+    const transferFunction = contractInterface.functions.find(func => {
+      const correctName = func.name === 'transfer';
+      const [amount, sender, recipient] = func.args;
+      const correctAmount = amount?.type === 'uint128';
+      const correctSender = sender?.type === 'principal';
+      const correctRecipient = recipient?.type === 'principal';
+      if (correctName) hasCorrectName = true;
+      if (correctSender) hasCorrectSender = true;
+      if (correctRecipient) hasCorrectRecipient = true;
+      if (correctAmount) hasCorrectAmount = true;
+      if (func.args.length === 3) hasCorrectNumberOfArgs = true;
```

Comment:
you are probably going to hate me for being the bearer of bad news, or at least of more work.  The `transfer` function was modified recently to include a 4th parameter (an optional memo with type `(optional (buff 34))`)

full details:
https://github.com/stacksgov/sips/pull/5#issuecomment-839001667
https://github.com/stacksgov/sips/pull/5/files#diff-b83747873c039e89a2dbcc593f0934ce59aaab9f58c8cacaefe4c01f6f2bdc95R52